### PR TITLE
Make iManufacturer configurable in factory flow

### PR DESCRIPTION
**Pull request by whitequark**
Fixes https://github.com/GlasgowEmbedded/glasgow/issues/347.

### CODE REVIEW

**Comment by whitequark** [comment]
Note to self: merge after #361

**Review by esden — APPROVED** [review]
This LGTM, I have tested this to provision a Glasgow and the iManufacturer is being set correctly.

File: software/glasgow/device/config.py [hunk 1]
```diff
@@ -80,7 +84,8 @@ def encode(self):
                            self.bitstream_size,
                            self.bitstream_id,
                            self.voltage_limit[0],
-                           self.voltage_limit[1])
+                           self.voltage_limit[1],
+                           self.manufacturer.encode("ascii"))
```
Comment (attie) [inline software/glasgow/device/config.py:88]: The default `None` makes this break... giving `b''` works.

```
Traceback (most recent call last):
  File "/usr/local/bin/glasgow", line 8, in <module>
    sys.exit(main())
  File "/opt/glasgow/software/glasgow/cli.py", line 876, in main
    exit(loop.run_until_complete(_main()))
  File "/usr/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/opt/glasgow/software/glasgow/cli.py", line 828, in _main
    fx2_config.append(0x4000 - GlasgowConfig.size, glasgow_config.encode())
  File "/opt/glasgow/software/glasgow/device/config.py", line 88, in encode
    self.manufacturer.encode("ascii"))
AttributeError: 'NoneType' object has no attribute 'encode'
```

**Review by attie — CHANGES_REQUESTED** [review]

File: software/glasgow/device/config.py (hunk 1 above)
Comment (whitequark) [reply software/glasgow/device/config.py:88]: Fixed

**Review by attie — APPROVED** [review]
Thank you!

**Pull request merged**